torch/distributed/checkpoint/state_dict_loader.py

```diff
@@ -215,3 +216,66 @@ def read_data():
         return None
 
     _ = distW.all_gather("read", read_data)
+
+
+def _load_state_dict_from_keys(
+    keys: Optional[set],
+    *,
+    checkpoint_id: Union[str, os.PathLike, None] = None,
+    storage_reader: Optional[StorageReader],
+    process_group: Optional[dist.ProcessGroup] = None,
+) -> Dict[str, Any]:
+    """
+    Load only the specified keys from the checkpoint.
```

Comment:
We should also mention that the checkpoint will not be sharded and all the ranks, if distributed is initialized, are going to load the full checkpoint.